### PR TITLE
BOAC-3247, server-side work to support Curated Group as Cohort filter option

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-3247

### CODE REVIEW

**Comment by johncrossman** [comment]
This does not include:
* delete cohort when delete dependent group
* update cohort SIDs (in db) when curated group is modified

Those changes will come next.

File: boac/externals/data_loch.py [hunk 1]
```diff
@@ -747,6 +749,16 @@ def get_students_query(     # noqa
...
         query_filter += f' AND sas.sid = ANY(:sids)'
         query_bindings.update({'sids': sids})
+    if curated_group_ids:
+        results = db.session.execute(
+            f'SELECT DISTINCT(sid) FROM student_group_members WHERE student_group_id = ANY(:curated_group_ids)',
+            {'curated_group_ids': curated_group_ids},
+        )
+        sids_of_curated_groups = []
+        for row in results:
+            sids_of_curated_groups.append(row['sid'])
```
Comment (paulkerschen) [inline boac/externals/data_loch.py]: @johncrossman Minor, but I think it’s more idiomatic (and perhaps runs more efficiently) to use a one-line list comprehension for the three above: `sids_of_curated_groups = [row[‘sid’] for row in results]`

**Comment by paulkerschen** [comment]
Looks good. Left one style q.

**Comment by johncrossman** [comment]
Thanks @paulkerschen  Amended. This PR also resolves https://jira.ets.berkeley.edu/jira/browse/BOAC-3251: If user has zero curated groups then disable, not hide, the menu option in 'Add new filter' dropdown.

**Comment by paulkerschen** [comment]
Screen Shot 2020-01-28 at 9 35 27 AM

**Pull request merged**